Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpAppxBuildTools.cs

```diff
@@ -484,6 +485,10 @@ public static void AddCapabilities(IBuildInfo buildInfo, XElement rootElement =
             var uwpBuildInfo = buildInfo as UwpBuildInfo;
 
             Debug.Assert(uwpBuildInfo != null);
+            if (uwpBuildInfo.DeviceCapabilities != null)
+            {
+                AddCapabilities(rootElement, uwpBuildInfo.DeviceCapabilities);
```

Comment:
So I'm not sure what happens with a manifest that has multiple instances of the same capability - i.e. if someone accidentally adds the gaze capability twice, or adds the wifiControl one twice.

It might instead be worth having something like

Set<string> capabilities = new Set<string>();
foreach (var capability in uwpBuildInfo.DeviceCapabilities)
{
   capabilities.add(capability)
}

And then instead of AddGazeInputCapability 

just do

capabilities.add("gazeInput");

Same for the research mode.

And then once we've gathered the set, then call AddCapabilities() - obviously changing it to take a set/collection instead of a list.

I think avoiding foot shooting would probably be good here!

Reply:
Sure, i'll give this a try.

Reply:
Actually since AddGazeInputCapability() is public I'll adjust it to use the collection mechanism internally

Reply:
The research mode function is a bit more complicated / different beyond capabilities so propose leaving this separate

Reply:
Hey @wiwei, it appears that AddCapability() already does a duplication check. I can still make this change, but believe it's redundant. Let me know!